ckan/lib/helpers.py

```diff
@@ -725,28 +726,46 @@ def get_facet_items_dict(facet, limit=None, exclude_active=False):
     facet -- the name of the facet to filter.
     limit -- the max. number of facet items to return.
     exclude_active -- only return unselected facets.
-
+    int_sort -- sort facets with int literals in 'name' by name and descending
     '''
-    if not c.search_facets or \
-            not c.search_facets.get(facet) or \
-            not c.search_facets.get(facet).get('items'):
+
+    def isint(v):
+        try:
+            i = int(v)
+        except:
+            i = v
+        return isinstance(i, int)
+
+    def sort(f):
+        if int_sort:
+            names = map(lambda it: it['display_name'], f)
+            all_ints = all(map(isint, names))
+            if all_ints:
+                return sorted(f, key=lambda it: -int(it['display_name']))
```

Comment:
You're first converting everything to `int` to see if it's an integer and then discard these integers only to compute them again later on. Why not do the conversion once (if `int_sort` is true) and use it for sorting, falling back to the default sort order if there's an error during conversion?